src/MultiFormats/HexString.cs

```diff
@@ -0,0 +1,161 @@
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace MultiFormats
+{
+    /// <summary>
+    ///   A codec for <see href="https://en.wikipedia.org/wiki/Hexadecimal">Hexadecimal</see>.
+    /// </summary>
+    /// <remarks>
+    ///   <para>
+    ///   A codec for a hexadecimal string, <see cref="Encode"/> and <see cref="Decode"/>.  Adds the extension method <see cref="ToHexString"/>
+    ///   to encode a byte array and <see cref="ToHexBuffer"/> to decode a hexadecimal <see cref="string"/>.
+    ///   </para>
+    /// </remarks>
+    public static class HexString
+    {
+        private static readonly string[] LowerCaseHexStrings =
+            Enumerable.Range(byte.MinValue, byte.MaxValue + 1)
+               .Select(v => v.ToString("x2"))
+               .ToArray();
+
+        private static readonly string[] UpperCaseHexStrings =
+            Enumerable.Range(byte.MinValue, byte.MaxValue + 1)
+               .Select(v => v.ToString("X2"))
+               .ToArray();
+
+        private static readonly Dictionary<string, byte> HexBytes =
+            Enumerable.Range(byte.MinValue, byte.MaxValue + 1)
+               .SelectMany(v => new[]
+                {
+                    new
+                    {
+                        Value = v, String = v.ToString("x2")
+                    },
+                    new
+                    {
+                        Value = v, String = v.ToString("X2")
+                    }
+                })
+               .Distinct()
+               .ToDictionary(v => v.String, v => (byte) v.Value);
+
+        /// <summary>
+        ///   Converts an array of 8-bit unsigned integers to its equivalent hexadecimal string representation.
+        /// </summary>
+        /// <param name="buffer">
+        ///   An array of <see cref="byte">8-bit unsigned integers</see>.
+        /// </param>
+        /// <param name="format">
+        ///   One of the format specifiers ("G" and "x" for lower-case hex digits, or "X" for the upper-case).
+        ///   The default is "G".
+        /// </param>
+        /// <returns>
+        ///   The string representation, in hexadecimal, of the contents of <paramref name="buffer"/>.  
+        /// </returns>
+        public static string Encode(byte[] buffer, string format = "G")
+        {
+            string[] hexStrings;
+            switch (format)
+            {
+                case "G":
+                case "x":
+                    hexStrings = LowerCaseHexStrings;
+                    break;
+                case "X":
+                    hexStrings = UpperCaseHexStrings;
+                    break;
+                default:
+                    throw new FormatException(
+                        string.Format("Invalid HexString format '{0}', only 'G', 'x' or 'X' are allowed.", format));
+            }
+
+            var s = new StringBuilder(buffer.Length * 2);
+            foreach (var v in buffer)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'foreach' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)